core/vmpu/src/vmpu.c

```diff
@@ -431,6 +463,13 @@ static void vmpu_enumerate_boxes(void)
         box_init(index, box_cfgtbl);
     }
 
+    if(debug_box_validated) {
+        for (int ii = 0; ii < UVISOR_MAX_BOXES; ii++) {
+            g_debug_interrupt_sp[ii] = g_context_current_states[ii].sp;
```

Comment:
1. This assignment cannot be done before g_debug_interrupt_sp[] is being initialized inside vmpu_configure_box_sram().
2. I agree. But this is not related to the PR. This can / should be done in another PR.